aws/resource_aws_security_group_rule.go

```diff
@@ -759,7 +759,19 @@ func setFromIPPerm(d *schema.ResourceData, sg *ec2.SecurityGroup, rule *ec2.IpPe
 		s := rule.UserIdGroupPairs[0]
 
 		if isVPC {
-			d.Set("source_security_group_id", s.GroupId)
+			if existingSourceSgId, ok := d.GetOk("source_security_group_id"); ok {
```

Comment:
looks good 👍 
as an alternative, you could even use the value of the `s.UserId` to handle the if/else as it'll be an empty string for security groups not tied to a specific accountId prefix
```suggestion
			if userId := aws.StringValue(s.UserId); len(userId) > 0 {
				d.Set("source_security_group_id", fmt.Sprintf("%s/%s", userId, aws.StringValue(s.GroupId)))
			} else {
				d.Set("source_security_group_id", s.GroupId)
			}
```

Reply:
Thanks for the review!

> as it'll be an empty string for security groups not tied to a specific accountId prefix

I believe that is not true. I believe `s.UserId` always has a value defined as the AWS API always returns a value for it whether or not we have a accoundId prefix or not in `source_security_group_id`

Reply:
😅 Using `d.Get()` or `d.GetOk()` to influence `d.Set()` calls is _generally_ discouraged since there are different semantics between resources created versus imported, however in this case, we don't have any options other than maybe #4454 (a `DiffSuppressFunc`) since the API seems to be very ambiguous about how it requires canonicalization. I'm hesitantly thinking this change to base the decision of `d.Set()` value is fine in principle since it should keep the existing resource behavior for previous configurations.